src/log/file_appender.cpp

```diff
@@ -36,6 +37,7 @@ namespace fc {
                   FC_ASSERT( cfg.rotation_limit >= cfg.rotation_interval );
 
                   rotate_files( true );
+                  delete_files();
```

Comment:
How about changing this to `_deletion_task = async(...)`?

Reply:
IMO it looks much cleaner like this. It shouldn't make much difference performance-wise because this call executes only once during startup.